src/ghc.jl

```diff
@@ -0,0 +1,133 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    GHC(kernel,merge_crit;k=35,vars=nothing)
+
+A method for partitioning spatial data into `k` clusters 
+using Geostatistical Hierarchical Clustering (GHC). This method generates
+clusters by assigning a unique cluster to each datum, 
+iteratively joining the two clusters which are most similar
+according to a kernel estimator on the data's spatial 
+dependence, and finally stopping this process at the step 
+where the clusters' silhouette index is optimized.
+
+## Parameters
+
+* `kernel`      - type of kernel used for estimator
+* `merge_crit`  - strategy to specify difference between clusters
+* `k`           - minimum number of observations contained in support 
+                    of kernel function centered at each point. 
+                    Defaults to 35.
+* `vars`        - Variables (or features) to consider (default to all)
+
+## References
+* F. Fouedjio. 2016. [A hierarchical clustering method for multivariate geostatistical data]
+  (https://www.sciencedirect.com/science/article/abs/pii/S2211675316300367)
+"""
+struct GHC <: PartitionMethod
+  kernel::AbstractString
+  merge_crit::AbstractString
+  k::Int
+  λ::Float64
+  vars::Union{Vector{Symbol},Nothing}
+end
+
+GHC(kernel::AbstractString, merge_crit::AbstractString; k=35,λ=10, vars=nothing) =
+  GHC(kernel, merge_crit, k,λ, vars)
+
+function partition(data::Data, method::GHC,kernel,merge_crit)
+  # variables used for clustering
+  dvars = Tables.schema(values(data)).names
+  vars = isnothing(method.vars) ? dvars : method.vars
+
+  @assert vars ⊆ dvars "GHC features not found in geospatial data"
+
+  kernels=["uniform,triangular","Epanechnikov","Gaussian"]
+  @assert kernel ⊆ kernels "Invalid kernel given"
+  merge_crits=[:single,:average,:complete]
+  @assert merge_crit ⊆ merge_crits "Invalid merge criterion given"
+    
+  # view subset of variables
+  ctor = constructor(typeof(data))
+  dom  = domain(data)
+  tab  = TableOperations.select(values(data), vars...)
+  Ω    = ctor(dom, Dict(paramdim(dom) => tab))
```

Comment:
Thanks for following the coding style. It really helps.

Reply:
Note, however that these assertions about kernel type and linkage function should be made earlier in the construction of the GHC object, not in the partition function.